src/fidesops/api/v1/endpoints/privacy_request_endpoints.py

```diff
@@ -577,25 +579,16 @@ def resume_privacy_request(
 
 
 def validate_manual_input(
-    manual_rows: List[Row], collection: CollectionAddress, db: Session
+    manual_rows: List[Optional[Row]],
+    collection: CollectionAddress,
+    dataset_graph: DatasetGraph,
 ) -> None:
     """Validate manually-added data for a collection.
 
     The specified collection must exist and all fields must be previously defined.
     """
-    datasets = DatasetConfig.all(db=db)
-    dataset_graphs = [dataset_config.get_graph() for dataset_config in datasets]
-    dataset_graph = DatasetGraph(*dataset_graphs)
-
-    node: Optional[Node] = dataset_graph.nodes.get(collection)
-    if not node:
-        raise HTTPException(
-            status_code=HTTP_422_UNPROCESSABLE_ENTITY,
-            detail=f"Cannot save manual rows. No collection in graph with name: '{collection.value}'.",
-        )
-
     for row in manual_rows:
-        for field_name in row:
+        for field_name in row or {}:
             if not dataset_graph.nodes[collection].contains_field(
                 lambda f: f.name == field_name  # pylint: disable=W0640
             ):
```

Comment:
> Your additional check is a different property, asserting that the individual row's key is non-null.
`row` is type `Row | None` and not `dict | None` right? So it would be checking for the presence of `Row` and not for a key. Or maybe I'm not understanding 😄 

Either way I like your idea of `List[Row]`, that seems clearer.

Reply:
Rows are dictionaries, so `for field_name in row` is looping through the dictionary keys, so it seemed like your `if not field_name` suggestion was checking if a key was None, which was different than the earlier check: `for field_name in row or {}` which is making sure Row is not None. 

I've adjusted to `List[Row]` though